### PR TITLE
Added Native Image documentation

### PR DESCRIPTION
**Pull request by shingjo**
Added documentation for native image.

### CODE REVIEW

**Review by alvarosanchez — APPROVED** [review]
Documenting this here is fine, but IMO we should also:

1. Add the version of `com.google.cloud:native-image-support` to the `micronaut-gcp-bom`.
2. Generate those build snippets in Micronaut Launch from the GCP feature if the GraalVM feature is also selected.

**Comment by shingjo** [comment]
> 2. Generate those build snippets in Micronaut Launch from the GCP feature if the GraalVM feature is also selected.
Will add this to my todo list.

**Review by melix — APPROVED** [review]

File: gradle.properties [hunk 1]
```diff
@@ -10,6 +10,7 @@ googleCloudCoreVersion=2.5.6
 googleFunctionsFrameworkApiVersion=1.0.4
 googleCloudPubsubVersion=1.116.0
 googleCloudSecretmanagerVersion=2.1.3
+googleCloudNativeImageSupportVersion=0.12.11
```
[the remark this answers is not in the document]
Comment (shingjo) [reply gradle.properties:13]: `0.13.1` Didn't work.

[the remark this answers is not in the document]
Comment (sdelamo) [reply gradle.properties:13]: ok, lets merged with 0.12.11 and we can upgrade to a version we know it works later on.

**Pull request merged**